net/nimble/host/src/ble_att.c

```diff
@@ -452,7 +452,7 @@ ble_att_chan_mtu(const struct ble_l2cap_chan *chan)
 
         mtu = BLE_ATT_MTU_DFLT;
     } else {
-        mtu = min(chan->my_mtu, chan->peer_mtu);
+        mtu = min(ble_att_preferred_mtu_val, chan->peer_mtu);
```

Comment:
Hi Łukasz,

Sorry if I am just being slow, but could you explain why this change (and the one below) is necessary?  I am envisioning this sequence:

1. We exchange MTU with peer: we specify 527, peer specifies 800; connection MTU is now 527.
2. Application sets preferred MTU to 100.

At this point, I would expect the connection MTU to be 527.  I believe your change causes the host to behave as if the MTU is 100.  We need to be prepared to receive 527-byte packets, since that is what we told the peer we support, so it seems the ble_att_chan_mtu() function should return 527.

But maybe I am misreading the code, or I forget how this works.

Thanks,
Chris